R/functions.R

```diff
@@ -185,75 +205,427 @@ getAtlasData <- function( experimentAccessions ) {
+}
 
-        message( "No species was provided. Will search for data from all available species." )
 
-    } else if( typeof( species ) != "character" ) {
+getNormalisedAtlasExpression <- function(experimentAccession, normalisation = "tpm") {
 
-        # Quit if the species isn't a character vector.
-        stop( "Please provide species as a character vector." )
+    # Ensure the experiment accession is in the correct format.
+    if (!.isValidExperimentAccession(experimentAccession)) {
+        stop("Experiment accession not valid. Cannot continue.")
+    }
 
-    } else if( length( species ) > 1 ) {
+    expType <- .getExperimentType(experimentAccession)
 
-        # Only allow one species to be specified.
-        stop( "More than one species found. You may only specify one species at a time." )
+    if (expType %in% c("rnaseq_mrna_baseline" )){
+        message(paste(experimentAccession," is ", expType , ", will continue downloading data"))
+    } else {
+        stop("Experiment type not contain normalised expression data.")
     }
 
-    # BioStudies API base URL.
-    bioAPIbase <- "http://www.ebi.ac.uk/biostudies/api/v1/search?query="
+    # Base URL for downloading data.
+    urlBase <- "ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/experiments"
 
-    # Construct the query URL
-    queryURL <- paste(
-        bioAPIbase,
-        paste( properties, collapse = "" ),
-        "&gxa=TRUE",  #"&link_type=gxa"
-        sep = ""
+    # Define filenames for configuration files.
+    configFile <- paste(experimentAccession, "-configuration.xml", sep = "")
+        
+    # XML URL for downloading config data.
+    xmlUrl <- paste(urlBase, experimentAccession, configFile, sep = "/")
+
+    if (normalisation %in% c("tpm", "fpkm")) {
+        # Ensure the normalisation is valid.
+        fileType <- match.arg(normalisation, c("tpm", "fpkm"))
+
+        # Define filenames for TPM and FPKM files.
+        expressionFile <- paste(experimentAccession, "-", fileType, "s.tsv", sep = "")
+
+        # Create full URLs for TPM and FPKM files.
+        expressionUrl <- paste(urlBase, experimentAccession, expressionFile, sep = "/")
+
+        # Initialise a list to hold the results.
+        results <- list()
+
+        # Download and read TPM file if requested or in "both" mode.
+        results <- .downloadExpressionFile(expressionUrl, experimentAccession)
+
+        # Remove NULL results for files that could not be downloaded.
+        results <- Filter(Negate(is.null), results)
+
+        # Check if any files were successfully downloaded.
+        if (length(results) == 0) {
+            stop("ERROR - Could not download the requested data. Please check the experiment type is RNAseq baseline or try again later.")
+        }
+
+        # Message indicating success.
+        if ("expression" %in% names(results)) {
+            message(paste("Successfully downloaded expression data for", experimentAccession))
+        }
```

Comment:
Does not apply anymore.. results stores file data directly not in expression key..